### ISSUE TITLE
Quick order: out of stock error handling is not the same between the cart page and quick order page

### ISSUE THREAD

**Issue by bgambocjaviniar** [SAP/spartacus#13641]
**Describe the bug**
Add an out of stock product from the cart page using the quick order 'quick access' form, and it throws a global message saying 'We could not find any products', when the product is out of stock.

**To Reproduce**
1. Go to 'cart page'
2. Add an out of stock product, such as 872912 in B2c using the quick access quick order form
3. See error

**Expected behavior**
The error should be handled similarly to the error handling on the quick order page.

**Screenshots**
![Screen Shot 2021-09-07 at 5 22 23 PM](https://user-images.githubusercontent.com/29099466/132412689-471589e7-5ccf-4aa4-b12e-7455406d794d.png)
![Screen Shot 2021-09-07 at 5 22 32 PM](https://user-images.githubusercontent.com/29099466/132412693-454bfbc8-3b4e-4791-906e-5b5dcb185227.png)

**Comment by dydome** [comment]
I do not agree with it. This is a different case. A scenario which you wrote, show us a situation where we are getting 400 error from API trying to add product. That is why we are showing an error global message, if you want to add a product that is already on the list with max stock level you will get a warning message. I am not sure we should implement a new message component on the cart as it is a huge change, rather I will leave it for 5.0 release where there is a plan to change current messages with the new one used on the quick order page, then it will look the same. @bgambocjaviniar @Xymmer

**Comment by Xymmer** [comment]
Nice improvement though, let's plan to do it but not urgent atm
